### PR TITLE
PERF-5359 use one selective predicate in MultiplannerWithGroup.yml workload

### PR DESCRIPTION
**Pull request by dstorch**
**Jira Ticket:** [PERF-5359](https://jira.mongodb.org/browse/PERF-5359)

### Whats Changed

This corrects an error with the original implementation of the workload. Before the change, the $match stage had no selective predicates. As a result, all of the documents in the collection matched, which in turn caused the workload to be dominated by query execution time. Since this is a multi-planning benchmark, we instead want the time required to select a plan to be significant relative to the overall query runtime.

This change also has the nice side effect of making the workload run much more quickly overall.

### Patch Testing Results

The workload seems to work as expected when I run it locally. I have this patch build created, but I'm figuring out how to take advantage of [DEVPROD-2159](https://jira.mongodb.org/browse/DEVPROD-2159) in order to only schedule the affected workload: https://spruce.mongodb.com/version/663a408fc899ea00071eed28/tasks

### CODE REVIEW

**Review by dpercy — APPROVED** [review]

**Comment by dstorch** [comment]
Thanks @dpercy! Looks like this is now waiting on @ghartnett before I can merge it.

My patch build completed successfully and the analysis against the base revision can be seen https://performance-analyzer.server-tig.prod.corp.mongodb.com/perf-analyzer-viz/?comparison_id=f104f085-dcb9-4051-af44-f063f2d85ebf. As expected, the performance of these benchmarks improved dramatically because the selectivity of the predicate improved dramatically.

**Review by jimoleary — APPROVED** [review]
LGTM can you let the build barons know that this will cause an improvement

**Comment by dstorch** [comment]
> LGTM can you let the build barons know that this will cause an improvement

Will do, thanks for the reminder.

**Pull request merged**
